apps/services/auth/ids-api/src/app/delegations/test/delegations-scopes.spec.ts

```diff
@@ -0,0 +1,174 @@
+import { Type } from '@nestjs/common'
+import { getConnectionToken } from '@nestjs/sequelize'
+import faker from 'faker'
+import { Sequelize } from 'sequelize-typescript'
+import request from 'supertest'
+
+import { DelegationType } from '@island.is/auth-api-lib'
+import { FixtureFactory } from '@island.is/services/auth/testing'
+import {
+  createCurrentUser,
+  createNationalId,
+} from '@island.is/testing/fixtures'
+import { TestApp, truncate } from '@island.is/testing/nest'
+
+import { setupWithAuth } from '../../../../test/setup'
+
+const domainName = faker.random.word()
+
+const identityResources = ['id1', 'id2']
+
+const legalGuardianScopes = ['lg1', 'lg2']
+const procurationHolderScopes = ['ph1', 'ph2']
+const customScopes1 = ['cu1', 'cu2']
+const customScopes2 = ['cu3', 'cu4']
+
+const apiScopes = [
+  ...legalGuardianScopes,
+  ...procurationHolderScopes,
+  ...customScopes1,
+  ...customScopes2,
+]
+
+const fromCustom = [
+  { nationalId: createNationalId('person'), scopes: customScopes1 },
+  { nationalId: createNationalId('person'), scopes: customScopes2 },
+  { nationalId: createNationalId('person'), scopes: [] },
+]
+
+interface TestCase {
+  fromNationalId: string
+  delegationType: DelegationType[]
+  expected: string[]
+}
+
+const testCases: Record<string, TestCase> = {
+  '1': {
+    fromNationalId: createNationalId('person'),
+    delegationType: [DelegationType.LegalGuardian],
+    expected: [...legalGuardianScopes, ...identityResources],
+  },
+  '2': {
+    fromNationalId: createNationalId('company'),
+    delegationType: [DelegationType.ProcurationHolder],
+    expected: [...procurationHolderScopes, ...identityResources],
+  },
+  '3': {
+    fromNationalId: fromCustom[0].nationalId,
+    delegationType: [DelegationType.Custom],
+    expected: [...customScopes1, ...identityResources],
```

Comment:
Could do this instead as you are using `fromCustom`

```suggestion
expected: [...fromCustom[0].scope, ...identityResources],
```